R/calcACOwnershipRates.R

```diff
@@ -0,0 +1,166 @@
+#'
+#' @importFrom magclass as.magpie
+#' @importFrom dplyr filter select rename mutate group_by slice_max ungroup coalesce anti_join full_join semi_join
+#' @importFrom tidyr as_tibble
+#' @importFrom quitte as.quitte
+#' @importFrom madrat toolCountryFill toolGetMapping
+#' @importFrom quitte interpolate_missing_periods
+
+calcACOwnershipRates <- function() {
+
+  # READ-IN DATA ---------------------------------------------------------------
+
+  ## Datasets ====
+
+  # household surveys
+  surveyData <- readSource("HouseholdSurveys", subtype = "appliances") %>%
+    as_tibble()
+
+  # IEA
+  ieaData <- readSource("IEA_acOwnership") %>%
+    as_tibble()
+
+  # Odyssee
+  odysseeData <- readSource("Odyssee") %>%
+    as.quitte(na.rm = TRUE)
+
+  # population (aggregation weights)
+  pop <- calcOutput("Population", scenario = "SSP2", aggregate = FALSE) %>%
+    as.quitte() %>%
+    removeColNa()
+
+
+  ## Mappings ====
+
+  regionmapIEA <- toolGetMapping("regionmappingIEA_acOwnership.csv",
+                                 type = "regional",
+                                 where = "mredgebuildings")
+
+  regionmapEDGE <- toolGetMapping("regionmappingISO-EDGE_EUR_ETP.csv",
+                                  type = "regional",
+                                  where = "mredgebuildings")
+
+
+
+  # PROCESS DATA ---------------------------------------------------------------
+
+  ## Process individual data sets ====
+
+  ### Household surveys ====
+
+  surveyData <- surveyData %>%
+    filter(!is.na(.data$value),
+           .data$variable == "ac") %>%
+    select(-"variable") %>%
+    rename("valueSurvey" = "value")
+
+
+  ### IEA ====
+
+  ieaData <- ieaData %>%
+    filter(!is.na(.data$value))
+
+  # filter out single country data points and give them priority
+  ieaSingleCountry <- ieaData %>%
+    filter(!.data$region %in% regionmapIEA$regionIEA) %>%
+    select(-"variable") %>%
+    rename("valueIEASingle" = "value") %>%
+    mutate(region = as.character(.data$region))
+
+  ieaSingleCountry$region <- toolCountry2isocode(ieaSingleCountry$region)
+
+  #TODO: filter regional values and disaggregate them to fill missing values? #nolint
+
+
+  ### Odyssee ====
+
+  # filter "AC equipment rate" and take most recent data point per region where necessary
+  odysseeData <- odysseeData %>%
+    filter(.data$variable == "teqcli") %>%
+    select("region", "period", "value") %>%
+    left_join(regionmapEDGE, by = c("region" = "CountryCode")) %>%
+    group_by(.data$region) %>%
+    (\(x) {
+      # split data into (non-)single-country regions
+      nonSingleCountry <- filter(x, .data$region != .data$RegionCodeEUR_ETP)
+      singleCountry    <- filter(x, .data$region == .data$RegionCodeEUR_ETP)
+      # take only max period for non-single-country regions
+      nonSingleCountry <- nonSingleCountry %>%
+        slice_max(order_by = .data$period, n = 1, with_ties = FALSE)
+      bind_rows(nonSingleCountry, singleCountry)
+    })() %>%
```

Comment:
I struggle a bit with this one. Why do you filter this way? Couldn't we also use all single-country region values to do the fit? And wouldn't this code do the same?
```suggestion
    filter(
      # non-single-country region: keep all periods
      .data$region != .data$RegionCodeEUR_ETP |
      # single-country region: only keep most recent period
      row_number() == which.max(.data$period)
    ) %>%
```